types/fs-base-ui/html_ui/JS/SimPlane.d.ts

```diff
@@ -0,0 +1,421 @@
+        FLIGHT_STATE_INTRO,
+        FLIGHT_STATE_PREFLIGHT_GATE,
+        FLIGHT_STATE_PREFLIGHT_PUSHBACK,
+        FLIGHT_STATE_PREFLIGHT_TAXI,
+        FLIGHT_STATE_PREFLIGHT_HOLDSHORT,
+        FLIGHT_STATE_FLIGHT_RUNWAY,
+        FLIGHT_STATE_FLIGHT_INITIAL_CLIMB,
+        FLIGHT_STATE_FLIGHT_CLIMB,
+        FLIGHT_STATE_FLIGHT_CRUISE,
+        FLIGHT_STATE_FLIGHT_DESCENT,
+        FLIGHT_STATE_JOINPLANE,
+        FLIGHT_STATE_LANDING_APPROACH,
+        FLIGHT_STATE_LANDING_FINAL,
+        FLIGHT_STATE_LANDING_TOUCHDOWN,
+        FLIGHT_STATE_LANDING_GROUNDROLL,
+        FLIGHT_STATE_LANDING_TAXI,
+        FLIGHT_STATE_LANDING_GATE,
+        FLIGHT_STATE_LANDING_REST,
+        FLIGHT_STATE_OUTRO,
+        FLIGHT_STATE_WAITING,
+        FLIGHT_STATE_TELEPORTTOSTATE,
+        FLIGHT_STATE_FREEFLIGHT,
+        FLIGHT_STATE_LANDINGCHALLENGE,
+        FLIGHT_STATE_BUSHTRIP,
+    }
+
+    enum FlightPhase {
+        FLIGHT_PHASE_PREFLIGHT,
+        FLIGHT_PHASE_TAXI,
+        FLIGHT_PHASE_TAKEOFF,
+        FLIGHT_PHASE_CLIMB,
+        FLIGHT_PHASE_CRUISE,
+        FLIGHT_PHASE_DESCENT,
+        FLIGHT_PHASE_APPROACH,
+        FLIGHT_PHASE_GOAROUND
+    }
+
+    enum ApproachType {
+        APPROACH_TYPE_UNKNOWN,
+        APPROACH_TYPE_VFR,
+        APPROACH_TYPE_HEL,
+        APPROACH_TYPE_TACAN,
+        APPROACH_TYPE_NDB,
+        APPROACH_TYPE_LORAN,
+        APPROACH_TYPE_RNAV,
+        APPROACH_TYPE_VOR,
+        APPROACH_TYPE_GPS,
+        APPROACH_TYPE_SDF,
+        APPROACH_TYPE_LDA,
+        APPROACH_TYPE_LOC,
+        APPROACH_TYPE_MLS,
+        APPROACH_TYPE_ILS
+    }
+
+    enum WorldRegion {
+        NORTH_AMERICA,
+        AUSTRALIA,
+        OTHER
+    }
+
+    enum NAV_AID_STATE {
+        OFF,
+        ADF,
+        VOR
+    }
+
+    enum NAV_AID_MODE {
+        NONE,
+        MANUAL,
+        REMOTE
+    }
+
+    namespace Simplane {
+        function getDesignSpeeds(): DesignSpeeds;
+        function getTrueSpeed(): Knots;
+        function getIndicatedSpeed(): Knots;
+        function getVerticalSpeed(): FeetPerMinute | null;
+        function getGroundSpeed(): Knots | null;
+        function getMachSpeed(): Mach | null;
+
+        /**
+         * Gets the V1 speed up during and before takeoff, -1 after.
+         */
+        function getV1Airspeed(): Knots | -1  | null;
+
+        /**
+         * Gets the VR speed up during and before takeoff, -1 after.
+         */
+        function getVRAirspeed(): Knots | -1  | null;
+
+        /**
+         * Gets the VR speed up during and before takeoff, -1 after.
+         */
+        function getV2Airspeed(): Knots | -1 | null;
+        function getREFAirspeed(): Knots | null;
+        function getVXAirspeed(): Knots | null;
+        function getFMCGreenDotSpeed(): number | null;
+        function getFMCApprGreenDotSpeed(): number | null;
+        function getGreenDotSpeed(): Knots | null;
+        function getCrossoverSpeedFactor(cas, mach: Mach): number | null;
```

Comment:
`cas` has no type.

Reply:
Using the JS code, I was unable to determine the type. Any suggestions?

Reply:
I'd guess that cas is the calibrated airspeed, which would be measured in Knots.